### PR TITLE
Added generate button to download csv teams

### PR DESCRIPTION
**Pull request by immmjimmy**

### CODE REVIEW

File: src/assets/js/pages/EventPage/tabs/ActionsTab.js [hunk 1]
```diff
@@ -28,6 +28,21 @@ export default class ActionsTab extends React.Component {
       });
   }
 
+  generateTeams = () => {
+    let eventAlias = this.props.event.alias;
+    exportUsers(eventAlias)
```
Comment (subhankar-panda) [inline src/assets/js/pages/EventPage/tabs/ActionsTab.js]: Great start - doesn't look like you're actually calling the API `generateTeams` though?

File: src/assets/js/data/Api.js [hunk 2]
```diff
@@ -251,6 +251,13 @@ export const exportUsers = (eventAlias) =>
     .set('Authorization', cookies.get(CookieTypes.admin.token, {path: '/'}))
     .use(apiPrefix);
 
+
+export const generateTeams = (eventAlias) =>
+  request
+    .get('admin/generateTeams/' + eventAlias)
+    .set('Authorization', cookies.get(CookieTyeps.admin.toke, {path: '/'}))
```
Comment (subhankar-panda) [inline src/assets/js/data/Api.js]: typos
```suggestion
    .set('Authorization', cookies.get(CookieTypes.admin.token, {path: '/'}))
```

File: src/server/routes/api.js [hunk 3]
```diff
@@ -138,6 +140,17 @@ module.exports = function(app) {
...
     });
 
+  api.get('/admin/generateTeams/:eventAlias', requireAuth, roleAuth(roles.ROLE_ADMIN),
+    isOrganiser, (req, res) => {
+      return User.find({event: req.event, deleted: false, sanitized: true})
+        .populate('account')
+        .exec()
+        .catch(err => Errors.respondError(res, err, Errors.DATABASE_ERROR))
+        .then((users) => {
+          return res.send(buildVertices(users));
```
Comment (subhankar-panda) [inline src/server/routes/api.js]: You'll have to do a little more here - look at https://github.com/UCSDTESC/Check-in/blob/master/src/server/routes/api.js#L132

File: src/server/routes/api.js [hunk 4]
```diff
@@ -138,6 +140,17 @@ module.exports = function(app) {
         });
     });
 
+  api.get('/admin/generateTeams/:eventAlias', requireAuth, roleAuth(roles.ROLE_ADMIN),
```
Comment (subhankar-panda) [inline src/server/routes/api.js]: change this to `/admin/exportTeams/:eventAlias` - we should use `generateTeams` for when we actually build the teams feature

**Comment by subhankar-panda** [comment]
Also looks like you committed a broken package-lock.json file, run `git rm package-lock.json` and `npm install` again before your next commit

File: src/assets/js/pages/EventPage/tabs/ActionsTab.js (hunk 1 above)
Comment (immmjimmy) [reply src/assets/js/pages/EventPage/tabs/ActionsTab.js]: Oops haha

File: src/assets/js/data/Api.js (hunk 2 above)
Comment (subhankar-panda) [reply src/assets/js/data/Api.js]: also wrap this in a `promisify(request......)` call - look at the other functions in this file

File: src/server/routes/helper.js [hunk 5]
```diff
@@ -201,6 +203,152 @@ function exportApplicantInfo(users, archive, finalize) {
   return csvStream.end();
 };
 
+function writeToCSV(masterList) {
+  console.log('\x1b[32m', '✓ Writing To teams.csv' ,'\x1b[0m');
```
Comment (RedbackThomson) [inline src/server/routes/helper.js]: Heroku logging does not support coloured outputs

File: src/server/routes/helper.js [hunk 6]
```diff
@@ -201,6 +203,152 @@ function exportApplicantInfo(users, archive, finalize) {
...
+    }
+    vertices[user.account.email.toLowerCase()] = currNodeChildren;
+
+    if (i === users.length - 1) {
+      return bfs(vertices);
+    }
+  });
+};
+
+
```
Comment (RedbackThomson) [inline src/server/routes/helper.js]: Might be better to have all of these in a separate file. This is less of an API helper method and more of its own system.

File: src/server/routes/helper.js [hunk 7]
```diff
@@ -209,6 +357,10 @@ module.exports = {
   isOrganiser,
   isSponsor,
   exportApplicantInfo,
+  writeToCSV,
+  cherryPick,
+  bfs,
+  buildVertices,
```
Comment (RedbackThomson) [inline src/server/routes/helper.js]: There is no need to export some of these functions, as they are never used outside of this file.

**Comment by RedbackThomson** [comment]
Invalidated by #134 .

**Pull request closed without merging**